libevmasm/Assembly.cpp

```diff
@@ -74,6 +74,117 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
+	int end = _json["end"].isInt() ?  _json["end"].asInt() : -1;
+	int srcIndex = _json["source"].isInt() ? _json["source"].asInt() : -1;
+	size_t modifierDepth = _json["modifierDepth"].isInt() ? static_cast<size_t>(_json["modifierDepth"].asInt()) : 0;
+	std::string value = _json["value"].isString() ? _json["value"].asString() : "";
+	std::string jumpType = _json["jumpType"].isString() ? _json["jumpType"].asString() : "";
+	solAssert(!name.empty(), "");
+
+	auto updateUsedTags = [&](u256 const& data) {
+		auto tag = static_cast<unsigned>(data);
+		if (m_usedTags <= tag)
+			m_usedTags = tag + 1;
+		return data;
+	};
+
+	auto immutableHash = [&](string const& _immutableName) -> h256 {
+		h256 hash(util::keccak256(value));
+		m_immutables[hash] = _immutableName;
+		return hash;
+	};
+
+	auto libraryHash = [&](string const& _libraryName) -> h256 {
+		h256 hash(util::keccak256(value));
+		m_libraries[hash] = _libraryName;
+		return hash;
+	};
+
+	SourceLocation location;
+	location.start = begin;
+	location.end = end;
+	if (srcIndex > -1 && srcIndex < static_cast<int>(sources().size()))
+		location.sourceName = sources()[static_cast<size_t>(srcIndex)];
+
+	AssemblyItem result(0);
+
+	if (c_instructions.find(name) != c_instructions.end())
+	{
+		AssemblyItem item{c_instructions.at(name), location};
+		item.m_modifierDepth = modifierDepth;
+		if (!jumpType.empty())
+			item.setJumpType(jumpType);
+		result = item;
+	}
+	else
+	{
+		if (name == "PUSH")
+		{
+			AssemblyItem item{AssemblyItemType::Push, u256("0x" + value), location};
+			if (!jumpType.empty())
+				item.setJumpType(jumpType);
+			result = item;
+		}
+		else if (name == "PUSH [ErrorTag]")
+			result = {AssemblyItemType::PushTag, 0, location};
+		else if (name == "PUSH [tag]")
+			result = {AssemblyItemType::PushTag, updateUsedTags(u256(value)), location};
+		else if (name == "PUSH [$]")
+			result = {AssemblyItemType::PushSub, u256("0x" + value), location};
+		else if (name == "PUSH #[$]")
+			result = {AssemblyItemType::PushSubSize, u256("0x" + value), location};
+		else if (name == "PUSHSIZE")
+			result = {AssemblyItemType::PushProgramSize, 0, location};
+		else if (name == "PUSHLIB")
+			result = {AssemblyItemType::PushLibraryAddress, libraryHash(value), location};
+		else if (name == "PUSHDEPLOYADDRESS")
+			result = {AssemblyItemType::PushDeployTimeAddress, 0, location};
+		else if (name == "PUSHIMMUTABLE")
+			result = {AssemblyItemType::PushImmutable, immutableHash(value), location};
+		else if (name == "ASSIGNIMMUTABLE")
+			result = {AssemblyItemType::AssignImmutable, immutableHash(value), location};
+		else if (name == "tag")
+			result = {AssemblyItemType::Tag, updateUsedTags(u256(value)), location};
+		else if (name == "PUSH data")
+			result = {AssemblyItemType::PushData, u256("0x" + value), location};
+		else if (name == "VERBATIM")
+		{
+			AssemblyItem item(fromHex(value), 0, 0);
+			item.setLocation(location);
```

Comment:
You could call `setLocation` at end end and just don't provide it in all the other constructors.